packages/auth-password/index.js

```diff
@@ -95,8 +96,20 @@ class PasswordAuthStrategy {
 
   async _matchItem(item, args, secretFieldInstance) {
     const { secretField } = this.config;
+    let hash = item[secretField];
     const secret = args[secretField];
-    return await secretFieldInstance.compare(secret, item[secretField]);
+    let message = `[passwordAuth:secret:mismatch] The ${secretField} provided is incorrect`;
+    if (!hash) {
```

Comment:
It looks like the logic has changed here? Previously we were accepting that `item[secretField]` contained the hash? Now if it does't exist we're generating a new one?

Reply:
`work factor` factored in by generating new hash, see line 77 for previous instance of such computation to prevent guess work for time based attack.